### PR TITLE
fix(docs): adjust api route documentation examples

### PR DESCRIPTION
**Pull request by bahag-buttf**
Fixed the example functions in the documentation to use `async` in order to use `await` within the function body.

### CODE REVIEW

**Review by manovotny — DISMISSED** [review]

**Comment by manovotny** [comment]
@bahag-buttf good catch! Thank you!

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 5c99344f179adf3d0580ab7802a1678797abe4c0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 5c99344f179adf3d0580ab7802a1678797abe4c0

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: docs/03-pages/01-building-your-application/01-routing/07-api-routes.mdx [hunk 1]
```diff
@@ -217,7 +217,7 @@ The following example sends a JSON response with the status code `200` (`OK`) an
 ```ts filename="pages/api/hello.ts" switcher
 import type { NextApiRequest, NextApiResponse } from 'next'
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
```
Comment (manovotny) [inline docs/03-pages/01-building-your-application/01-routing/07-api-routes.mdx:220]:
```suggestion
export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
```

File: docs/03-pages/01-building-your-application/01-routing/07-api-routes.mdx [hunk 2]
```diff
@@ -247,7 +247,7 @@ The following example sends a HTTP response with the status code `200` (`OK`) an
...
```
Comment (manovotny) [inline docs/03-pages/01-building-your-application/01-routing/07-api-routes.mdx:250]:
```suggestion
export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
```

File: docs/03-pages/01-building-your-application/01-routing/07-api-routes.mdx [hunk 3]
```diff
@@ -277,7 +277,7 @@ The following example redirects the client to the `/` path if the form is succes
...
```
Comment (manovotny) [inline docs/03-pages/01-building-your-application/01-routing/07-api-routes.mdx:280]:
```suggestion
export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
```

**Review by manovotny — APPROVED** [review]

**Pull request merged**